interface/src/app/gene/network/network.js

```diff
@@ -17,6 +18,23 @@ angular.module('adage.gene.network', [
         controller: 'GeneNetworkCtrl as ctrl'
       }
     },
+    // When "gene_network" state exits, remove "gene-tip" and "edge-tip"
+    // elements from the DOM. (Without this function, the tips window will
+    // be visible in other states too.)
+    onExit: function() {
+      var removeTips = function(className) {
+        var elements = document.getElementsByClassName(className);
+        var i, n;
+        if (elements) { // This should be always true.
+          n = elements.length;
+          for (i = 0; i < n; ++i) {
+            elements[i].parentNode.removeChild(elements[i]);
+          }
+        }
+      };
+      removeTips('gene-tip');
+      removeTips('edge-tip');
+    },
```

Comment:
I plan to write more unit tests and E2E tests after this deadline. I have put it on the list of "Low priority tasks" on the shared google doc.